internal/server/iigointernal/legislature.go

```
@@ -29,53 +30,54 @@ func (l *legislature) loadClientSpeaker(clientSpeakerPointer roles.Speaker) {
+// sendJudgeSalary conduct the transaction based on amount from client implementation
+func (l *legislature) sendJudgeSalary() error {
 	if l.clientSpeaker != nil {
-		amount, judgePaid := l.clientSpeaker.PayJudge(l.judgeSalary)
-		if judgePaid {
-			judicialBranch.budget = amount
+		amountReturn := l.clientSpeaker.PayJudge(l.judgeSalary)
+		if amountReturn.ActionTaken {
+			// Subtract from common resources pool
+			amountWithdraw, withdrawSuccess := WithdrawFromCommonPool(amountReturn.JudgeSalary, l.gameState)
+
+			if withdrawSuccess {
+				// Pay into the client private resources pool
+				depositIntoClientPrivatePool(amountWithdraw, l.gameState.JudgeID, l.gameState)
+			}
 		}
-		return
 	}
-	judicialBranch.budget = l.judgeSalary
+	return errors.Errorf("Cannot perform sendJudgeSalary")
 }
 
 // Receive a rule to call a vote on
-func (l *legislature) setRuleToVote(r string) {
-	ruleToBeVoted, ruleSet := l.clientSpeaker.DecideAgenda(r)
-	if ruleSet {
-		l.ruleToVote = ruleToBeVoted
+func (l *legislature) setRuleToVote(r string) error {
+
+	if !l.incurServiceCharge(actionCost.SetRuleToVoteActionCost) {
+		return errors.Errorf("Insufficient Budget in common Pool: setRuleToVote")
+	}
+
+	agendaReturn := l.clientSpeaker.DecideAgenda(r)
+	if agendaReturn.ActionTaken {
+		l.ruleToVote = agendaReturn.RuleID
 	}
+	return nil
 }
 
 //Asks islands to vote on a rule
 //Called by orchestration
-func (l *legislature) setVotingResult(clientIDs []shared.ClientID) {
+func (l *legislature) setVotingResult(clientIDs []shared.ClientID) error {
 
-	ruleID, participatingIslands, voteDecided := l.clientSpeaker.DecideVote(l.ruleToVote, clientIDs)
-	if !voteDecided {
-		return
+	if !l.incurServiceCharge(actionCost.SetVotingResultActionCost) {
+		return errors.Errorf("Insufficient Budget in common Pool: setVotingResult")
+	}
+	returnVote := l.clientSpeaker.DecideVote(l.ruleToVote, clientIDs)
+	if !returnVote.ActionTaken {
+		return nil
```

Do we need to return an error here?